modules/heat_conduction/src/kernels/AnisoHeatConduction.C

```diff
@@ -23,53 +23,43 @@ AnisoHeatConduction::validParams()
 AnisoHeatConduction::AnisoHeatConduction(const InputParameters & parameters)
   : Kernel(parameters), _dim(_subproblem.mesh().dimension())
 {
-  _k_i[0] = _k_i[1] = _k_i[2] = NULL;
-  _k_i_dT[0] = _k_i_dT[1] = _k_i_dT[2] = NULL;
+  _k = nullptr;
+  _k_dT = nullptr;
 
-  if (hasMaterialProperty<Real>("thermal_conductivity_x"))
-    _k_i[0] = &getMaterialProperty<Real>("thermal_conductivity_x");
+  if (hasMaterialProperty<RankTwoTensor>("thermal_conductivity"))
+    _k = &getMaterialProperty<RankTwoTensor>("thermal_conductivity");
 
-  if (hasMaterialProperty<Real>("thermal_conductivity_x_dT"))
-    _k_i_dT[0] = &getMaterialProperty<Real>("thermal_conductivity_x_dT");
+  if (hasMaterialProperty<RankTwoTensor>("thermal_conductivity_dT"))
+    _k_dT = &getMaterialProperty<RankTwoTensor>("thermal_conductivity_dT");
```

Comment:
would be nice to use the `DerivativeMaterialInterface` here

Reply:
Done